workshop/session_3/09_biopython_sequences.ipynb

```diff
@@ -0,0 +1,856 @@
+    "editable": false
+   },
+   "source": [
+    "The reversal of the process - getting from RNA to DNA is also supported."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 138,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Transcibed:       ACUG\n",
+      "Back transcribed: ACTG\n"
+     ]
+    }
+   ],
+   "source": [
+    "transcribed = dna_sequence.transcribe()\n",
+    "print(\"Transcibed:      \", transcribed)\n",
+    "print(\"Back transcribed:\", transcribed.back_transcribe())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "editable": false
+   },
+   "source": [
+    "**Biopython** makes translation from RNA (or directly DNA) to aminoacid sequence a very easy task. \n",
+    "\n",
+    "*Note that different codon tables are also supported.*"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 139,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "Seq('TRR')"
+      ]
+     },
+     "execution_count": 139,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "sequence = Seq(\"ACGCGACGA\")\n",
+    "sequence.translate()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "editable": false
+   },
+   "source": [
+    "### Using `Seq` methods directly on `string`s. \n",
+    "It is possible to ignore `Seq` class abstraction and import and execute relevant methods directly on simple `string`s. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 140,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Transcription directly from 'ACGT' string, without using Seq object: ACGU\n"
+     ]
+    }
+   ],
+   "source": [
+    "from Bio.Seq import transcribe\n",
+    "print(\"Transcription directly from 'ACGT' string, without using Seq object:\", transcribe(\"ACGT\"))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "editable": false
+   },
+   "source": [
+    "## Alignment\n",
+    "\n",
+    "One of the most common uses of bioinformatics tools is sequence alignment. \n",
+    "**Biopython** provides support for various command-line and online alignment tools. The details can be found [here](https://biopython.org/docs/latest/api/Bio.Align.Applications.html). \n",
+    "\n",
+    "For this section of this workshop, we will use the built-in, out-of-the-box alignment method contained in `pairwise2` module. It allows aligning 2 sequences at a time.\n",
+    "\n",
+    "We will first import it, run a global (full sequence) alignment and present relevant results."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 141,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Alignment(seqA='ACGT-', seqB='ACG-C', score=3.0, start=0, end=5)\n",
```

Comment:
What does 'score' mean in the alignment? Maybe add this to the text?

Reply:
Explained in the preceding cell.